lib/isZodErrorLike.ts

```diff
@@ -0,0 +1,5 @@
+import type * as zod from 'zod';
+
+export function isZodErrorLike(err: unknown): err is zod.ZodError {
+  return err instanceof Error && err.name === 'ZodError';
```

Comment:
suggestion: As discussed offline, we could also introduce a further check for the `issues` prop inside `err`. That would significantly narrow down the false positive cases where non-zod custom errors are named as `ZodError`.

Reply:
Fixed with [3ed91ed](https://github.com/causaly/zod-validation-error/pull/304/commits/3ed91edb86c19804df99cb1e17c3e01e9640907b)